extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/DevUICORSFilter.java

```diff
@@ -46,18 +50,42 @@ public void handle(RoutingContext event) {
         HttpServerRequest request = event.request();
         HttpServerResponse response = event.response();
         String origin = request.getHeader(HttpHeaders.ORIGIN);
-        if (origin == null) {
-            corsFilter().handle(event);
-        } else {
-            if (origin.startsWith(HTTP_LOCAL_HOST) || origin.startsWith(HTTPS_LOCAL_HOST)
-                    || origin.startsWith(HTTP_LOCAL_HOST_IP) || origin.startsWith(HTTPS_LOCAL_HOST_IP)) {
+        if (hostIsValid(event)) {
+            if (origin == null) {
                 corsFilter().handle(event);
             } else {
-                LOG.errorf("Only localhost origin is allowed, but Origin header value is: %s", origin);
-                response.setStatusCode(403);
-                response.setStatusMessage("CORS Rejected - Invalid origin");
-                response.end();
+                if (origin.startsWith(HTTP_LOCAL_HOST) || origin.startsWith(HTTPS_LOCAL_HOST)
```

Comment:
What happens if cors are disabled? 

I've been working on a setup where we need to disable the dev-ui CORS (I understand the risk).

Reply:
Ok I moved this logic to a dedicated filter